browser/tor/tor_profile_service_factory.cc

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/tor/tor_profile_service_factory.h"
+
+#include "brave/browser/tor/tor_profile_service_impl.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+
+// static
+tor::TorProfileService* TorProfileServiceFactory::GetForProfile(
+    Profile* profile) {
+  if (profile->IsOffTheRecord())
+    return NULL;
```

Comment:
nit: prefer nullptr since that's the standard.

Brians-iMac-Pro-2:src brianbondy$ git grep return\ NULL | wc -l
    3843
Brians-iMac-Pro-2:src brianbondy$ git grep return\ nullptr | wc -l
   12074